torchvision/io/_video_opt.py

```diff
@@ -517,7 +531,7 @@ def get_pts(time_base):
     return vframes, aframes, _info
 
 
-def _read_video_timestamps(filename, pts_unit="pts"):
+def _read_video_timestamps(filename: str, pts_unit: str = "pts") -> Tuple[List[Fraction], Optional[float]]:
```

Comment:
```suggestion
def _read_video_timestamps(
        filename: str, pts_unit: str = "pts"
) -> Tuple[Union[List[int], List[Fraction]], Optional[float]]:
```

In addition, you need to put `pts: Union[List[int], List[Fraction]]` before the first assignment of it. Doing that, we can drop the ignores.